### ISSUE TITLE
Accessibility: A11y_Live Chat Widget_Chat Window_AI4W: Ensures all ARIA attributes have valid values (#webchat__toaster__list__gf1z1)

### ISSUE THREAD

**Issue by compulim** [microsoft/BotFramework-WebChat#3053]
Copied from #1749559.

![1749559_A11y_Live Chat Widget_Chat Window_AI4W](https://user-images.githubusercontent.com/1622400/77822067-908bfa80-70ac-11ea-8ed2-96fb04c9169c.png)

“Check out Accessibility Insights! - Identify accessibility bugs before check-in and make bug fixing faster and easier.”

Environment Details:

Application Name: Live Chat Widget

#URL:https://xxx.blob.core.windows.net/portals/democ2occatest4.html

Browser Versions: Microsoft Edge Version 80.0.361.54 (Official build) (64-bit)

Windows Version: Windows10

Chrome Version: Version 80.0.3987.132 (Official Build) (64-bit)

 

Repro Steps: ​​

 

1)Launch the application url : https://xxx.blob.core.windows.net/portals/democ2occatest4.html

2)welcome to live chat support Screen get open.

4)Navigate to "Chat Now button" and click on it.

5)"Chat Form" screen get open.

6)Fill all the mandatory detail and then Click on "Submit" Button

7)now "Chat Window" get open.

8)Run AI4W and check the issue.

 

Issue: Ensures all ARIA attributes have valid values (aria-valid-attr-value - https://dequeuniversity.com/rules/axe/3.5/aria-valid-attr-value?application=msftAI)

 

 

Target application: Test Page For Common Scenarios - https://xxx.blob.core.windows.net/portals/democ2occatest4.html

 

 

Element path: #Microsoft_Omnichannel_LCWidget_Chat_Iframe_Window;#webchat__toaster__list__gf1z1

 

Snippet: <ul aria-labelledby="webchat__toaster__header__crqlh" class="webchat__toaster__list" id="webchat__toaster__list__gf1z1" role="region"></ul>

 

How to fix:

Fix all of the following:

  Invalid ARIA attribute value: aria-labelledby="webchat__toaster__header__crqlh"

 

Environment: Chrome version 80.0.3987.149

 

====

 

This accessibility issue was found using Accessibility Insights for Web 2.15.0 (axe-core 3.5.1), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.

 

Recommendations:

Refer below link which is repository of bug fixes code snippets:

https://microsoft.sharepoint.com/teams/msenable/mas/pages/browse-fixes.aspx

 

MAS Reference:

 

MAS 4.1.1:Parsing

https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={2e2a1540-aaf7-48a0-a53e-8f10cf309999}

 

MAS 4.1.2:

https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={248054a6-5e68-4771-9e1e-242fb5025730}

**Comment by compulim** [comment]
The issue mentioned this aria attribute, `aria-labelledby="webchat__toaster__header__crqlh"`.

Will need to investigate, but the value in `aria-labelledby` should be an ID of the element, which will probably be a random string.

Will investigate if the issue is talking about the element referenced by `aria-labelledby` is missing, or if that element contains no good alt text.

**Comment by compulim** [comment]
This is dupe of #3054.

**Comment by nathanlindorff** [comment]
@compulim I'm not so sure this is a complete dup of 3054. I'm getting the same issue when running cypress tests on our bot in webchat, and the issue seems to be the fact that it doesn't like the role of 'region' on a 'ul' element.

Here is the console output from the cypress error inspector

Command:      a11y error!
Id:           aria-allowed-role
Impact:       minor
Tags:         Array(2)0: "cat.aria"1: "best-practice"length: 2__proto__: Array(0)
Description:  Ensures role attribute has an appropriate value for the element
Help:         ARIA role must be appropriate for the element
HelpUrl:      https://dequeuniversity.com/rules/axe/3.5/aria-allowed-role?application=axeAPI

Nodes:        
```
[
    {
        "any": [
            {
                "id": "aria-allowed-role",
                "data": [
                    "region"
                ],
                "relatedNodes": [],
                "impact": "minor",
                "message": "ARIA role region is not allowed for given element"
            }
        ],
        "all": [],
        "none": [],
        "impact": "minor",
        "html": "<ul aria-labelledby=\"webchat__toaster__header__hmrvl\" class=\"webchat__toaster__list\" id=\"webchat__toaster__list__cg4t8\" role=\"region\"></ul>",
        "target": [
            "#webchat__toaster__list__cg4t8"
        ],
        "failureSummary": "Fix any of the following:\n  ARIA role region is not allowed for given element"
    }
]
```
